### PR TITLE
Fix the Linux broker testing job in the daily pipeline

### PR DESCRIPTION
**Pull request by tanmaymanolkar1**
WHAT?
Fix the Linux broker testing in the Auth Client Android Daily Release Pipeline : https://identitydivision.visualstudio.com/Engineering/_build?definitionId=1515&_a=summary

We have also added the testParams, specifically for test jobs

A successful run: 
https://identitydivision.visualstudio.com/Engineering/_build/results?buildId=1101900&view=results

https://identitydivision.visualstudio.com/Engineering/_build/results?buildId=1102421&view=logs&j=03c4f6f1-88e4-5ead-1f97-f50c8da9a6d0&t=dcaefdc0-41bf-573b-6391-81b60e4bdc1c

ADO Item: https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2574910

### CODE REVIEW

File: azure-pipelines/continuous-delivery/assemble&publish.yml [hunk 1]
```diff
@@ -63,12 +70,14 @@ jobs:
         or( eq(${{ parameters.shouldPublish }}, 'True'), eq( variables['Build.Reason'], 'Schedule'))
       )
     displayName: Publish ${{ parameters.project }}
+    enabled: false
```
Comment (iamgusain) [inline azure-pipelines/continuous-delivery/assemble&publish.yml:73]: remove temp changes before merging

**Review by iamgusain — CHANGES_REQUESTED** [review]
Remove temporary/test changes before mergin

File: azure-pipelines/continuous-delivery/assemble&publish.yml (hunk 1 above)
Comment (tanmaymanolkar1) [reply azure-pipelines/continuous-delivery/assemble&publish.yml]: removed

**Comment by tanmaymanolkar1** [comment]
> Remove temporary/test changes before mergin

ack

File: azure-pipelines/continuous-delivery/assemble&publish.yml [hunk 2]
```diff
@@ -26,7 +26,12 @@ parameters:
 - name: shouldRunUnitTests
   type: boolean
   default: True
-
+- name: shouldRunLinuxInlineScript
+  type: boolean
+  default: False
+- name: testParams
```
Comment (fadidurah) [inline azure-pipelines/continuous-delivery/assemble&publish.yml:32]: I think renaming this to something like "linuxTestParams" is more clear

**Review by fadidurah — APPROVED** [review]

File: azure-pipelines/continuous-delivery/assemble&publish.yml (hunk 2 above)
Comment (iamgusain) [reply azure-pipelines/continuous-delivery/assemble&publish.yml:32]: I think testParams is fine, as we can also use it for other projects if needed

File: azure-pipelines/continuous-delivery/assemble&publish.yml [hunk 3]
```diff
@@ -87,7 +92,40 @@ jobs:
...
+        script: |
+          sudo apt-get install -y dbus-x11
+          sudo apt-get install -y dos2unix
+          dos2unix gradlew
+          chmod +x gradlew
+          export DISPLAY=:0.0
+          eval $(dbus-launch --sh-syntax)
+          sudo apt install gnome-keyring
+          /usr/bin/gnome-keyring-daemon --start --components=secrets
+          ./gradlew LinuxBroker:linuxBrokerUnitTestCoverageReport ${{ parameters.testParams}}
```
Comment (iamgusain) [inline azure-pipelines/continuous-delivery/assemble&publish.yml:112]: Do we need this to run test here?

File: azure-pipelines/continuous-delivery/assemble&publish.yml [hunk 4]
```diff
@@ -87,7 +92,40 @@ jobs:
...
+      enabled: ${{ parameters.shouldRunUnitTests }}
+      continueOnError: true
+      inputs:
+        tasks: ${{ parameters.project }}:${{ parameters.testCmd}} ${{ parameters.testParams}}
+        testRunTitle: '${{ parameters.project }}_UnitTests'
+      env:
+        ${{ parameters.vstsMvnAndroidUsername }}: $(AndroidAuthClient-Internal-Maven-Username)
+        ${{ parameters.vstsMvnAndroidAccessToken }}: $(System.AccessToken)
     - task: Gradle@3
+      condition: ne(${{ parameters.shouldRunLinuxInlineScript }}, 'True')
```
Comment (iamgusain) [inline azure-pipelines/continuous-delivery/assemble&publish.yml:128]: Why separate task for Linux?

File: azure-pipelines/continuous-delivery/assemble&publish.yml [hunk 5]
```diff
@@ -26,7 +26,12 @@ parameters:
 - name: shouldRunUnitTests
   type: boolean
   default: True
-
+- name: shouldRunLinuxInlineScript
```
Comment (p3dr0rv) [inline azure-pipelines/continuous-delivery/assemble&publish.yml:29]: IMO we should get rid of this variable and just check if the project is LinuxBroker

File: azure-pipelines/continuous-delivery/assemble&publish.yml (hunk 3 above)
Comment (p3dr0rv) [reply azure-pipelines/continuous-delivery/assemble&publish.yml:112]: yep, If this is just a set up for Linux broker we should remove this line

Comment (tanmaymanolkar1) [reply azure-pipelines/continuous-delivery/assemble&publish.yml:112]: yeah we need to to activate the keyring, I tried removing it and the test cases they fail in the separate testing job

File: azure-pipelines/continuous-delivery/assemble&publish.yml (hunk 2 above)
Comment (tanmaymanolkar1) [reply azure-pipelines/continuous-delivery/assemble&publish.yml:32]: yes testParams is more for a generic use

File: azure-pipelines/continuous-delivery/assemble&publish.yml (hunk 5 above)
Comment (tanmaymanolkar1) [reply azure-pipelines/continuous-delivery/assemble&publish.yml:29]: this parameter particularly set to make sure we run the script only when needed, we have other jobs like Linux Broker publishing where we wouldnt want the associated script to run.

File: azure-pipelines/continuous-delivery/assemble&publish.yml [hunk 6]
```diff
@@ -87,7 +92,40 @@ jobs:
         KeyVaultName: 'ADALTestInfo'
         SecretsFilter: 'AndroidAutomationRunnerAppSecret'
         RunAsPreJob: false
+    - task: Bash@3
+      condition: eq(${{ parameters.shouldRunLinuxInlineScript }}, 'True')
```
[the remark this answers is not in the document]
Comment (tanmaymanolkar1) [reply azure-pipelines/continuous-delivery/assemble&publish.yml:96]: I have added a separate flag to make sure we can separate out specific tasks in Linux Broker, it is a better to have flag which we can mend to particularly run the associated scripts.

File: azure-pipelines/continuous-delivery/assemble&publish.yml (hunk 4 above)
Comment (tanmaymanolkar1) [reply azure-pipelines/continuous-delivery/assemble&publish.yml:128]: I will get the tasks merged

**Comment by tanmaymanolkar1** [comment]
> 

Updated

**Comment by tanmaymanolkar1** [comment]
> BTW I think the test task is incorrect. it should use ${{ parameters.testParams}}
> 
> tasks: ${{ parameters.project }}:${{ parameters.testCmd}} _${{ parameters.assembleParams}}_ -Psugar=true -PlabSecret=$(AndroidAutomationRunnerAppSecret) -PshouldSkipLongRunningTest=true -PcodeCoverageEnabled=true

Updated

**Review by p3dr0rv — APPROVED** [review]

**Comment by tanmaymanolkar1** [comment]
> Remove temporary/test changes before mergin

Noted

File: azure-pipelines/continuous-delivery/assemble&publish.yml (hunk 2 above)
Comment (fadidurah) [reply azure-pipelines/continuous-delivery/assemble&publish.yml:29]: Ahh I see. When I first looked at it it was only being used under the Linux script condition. Now that's used for all the params, this naming is fine.

**Pull request merged**